File: Reference-Android-Java/app/src/main/java/xyz/fay/reference/feature/city/CityViewModel.java
package xyz.fay.reference.feature.city;

import android.content.Context;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import xyz.fay.reference.networking.NetworkManager;
import xyz.fay.reference.networking.RequestHandler;
import xyz.fay.reference.networking.response.CityResponse;

public class CityViewModel extends ViewModel {
    private final MutableLiveData<CityResponse> cityResponse = new MutableLiveData<>();

    public void fetchData(Context context) {
        NetworkManager manager = new NetworkManager();
        manager.getCity(context, new RequestHandler<CityResponse>() {
            @Override
            public void completion(CityResponse response) {
                cityResponse.setValue(response);
            }
        });
//        manager.getCity(context, response -> {
//            cityResponse.setValue(response);
//        });
//        manager.getCity(context, cityResponse::setValue);
    }

    public MutableLiveData<CityResponse> getCityResponse() {
        return cityResponse;
    }
}

File: Reference-Android-Kotlin/app/src/main/java/xyz/fay/reference/feature/city/CityViewModel.kt
package xyz.fay.reference.feature.city

import android.content.Context
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import xyz.fay.reference.networking.NetworkManager
import xyz.fay.reference.networking.response.CityResponse

class CityViewModel : ViewModel() {
    private val _cityResponse = MutableLiveData<CityResponse>()
    val cityResponse: MutableLiveData<CityResponse>
        get() = _cityResponse

    fun fetchData(context: Context) {
        val manager = NetworkManager()
        manager.getCity(context) {
            _cityResponse.value = it
        }
//        manager.getCity(context, _cityResponse::setValue)
    }
}
